lib/elixir/pages/Naming Conventions.md

```diff
@@ -0,0 +1,90 @@
+# Naming Conventions
+
+## Naming modules
+
+Elixir module names are written in "upper camel case", like `OptionParser`.
+
+Capital letters are kept in abbreviations, like `ExUnit.CaptureIO` or `Mix.SCM`.
```

Comment:
While this is the general convention, you sometimes have to violate it when defining mix tasks.  We found that when we defined a mix task module as `defmodule Mix.Tasks.MyApp.CLI` it was not invokable from the command line via `mix my_app.cli` but when we named the module `Mix.Tasks.MyApp.Cli` it was.  The problem is that mix converts `cli` to `Cli`, not knowing it's an acronym.

Maybe it's worth saying something about this?  Alternately, I wonder if we can make mix smarter so that it uses `Mix.Tasks.MyApp.CLI` if it exists but `Mix.Tasks.MyApp.Cli` does not?